pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -254,8 +271,20 @@ func (i *inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentr
 		return nil, syserror.EOVERFLOW
 	}
 
+	var fd *fileDescription
+	var fdImpl vfs.FileDescriptionImpl
+
 	// FOPEN_KEEP_CACHE is the defualt flag for noOpen.
-	fd := fileDescription{OpenFlag: linux.FOPEN_KEEP_CACHE}
+	if isDir {
+		fd := &fileDescription{OpenFlag: linux.FOPEN_KEEP_CACHE}
```

Comment:
are you sure that we need to re-declare fd here? Then fd is used on the 295 line, but after this change it will be nil.